Security/Security-with-AAD-pod-managed-identities/README.md

```diff
@@ -0,0 +1,279 @@
+
+In this section, we will be focusing on **Managed Identities** which is a functionality that makes authentication with a service principal easier. It works by assigning an identity to a compute resource on Azure such as a Virtual Machine (VM) or Azure Functions. These resources can authenticate using that managed identity through an endpoint that is only accessible from that machine. 
+
+**Azure AD pod-managed identities** allows you to assign managed identities to pods in Kubernetes. Since pods in Kubernetes run on VMs, by default, each pod would be able to access the managed identity endpoint and authenticate using that identity. Using Azure AD pod-managed identities, pods can no longer reach the internal endpoint designated for the VM and would rather only be able to get access through identities assigned to that specific pod.
+
+In this section we will be going over the following:
+
+1. Updating your cluster so that it has Azure AD pod-managed identities feature 
+1. Linking an identity to your cluster
+1. Using a pod with managed identity
+
+For more indepth details of how AAD pod-managed identities work please refer to Chapter 9 of the third edition of the book [Hands-on Kubernetes on Azure](https://www.packtpub.com/product/hands-on-kubernetes-on-azure-third-edition/9781801079945). 
+
+## Turning on Azure AD pod-managed identity feature in AKS cluster
+
+1. Assuming you have the environment variables defined and the AKS preview feature added for Azure CLI in the setup section of this workshop, Register the `EnablePodIdentityPreview` feature
+
+   > Note that for this section, you need to be using an Azure tenant with AAD instance where you have global administrator permissions. If you don't have one please create [a free account here](https://azure.microsoft.com/en-ca/free/search/?OCID=AID2100017_SEM_64fae88fcd6f1197e6ee88294daa302c:G:s&ef_id=64fae88fcd6f1197e6ee88294daa302c:G:s&msclkid=64fae88fcd6f1197e6ee88294daa302c) and deploy an AKS cluster **with Azure CNI** before preceding. It doesn't have to be the AKS workshop cluster.
+
+   ```bash
+   az feature register --name EnablePodIdentityPreview \
+   	--namespace Microsoft.ContainerService
+   ```
+
+1. Update the fruit smoothie AKS cluster with Azure CNI to include pod-managed identity. 
+
+   ```bash
+   az aks update -g $RESOURCE_GROUP -n $AKS_CLUSTER_NAME --enable-pod-identity
+   ```
+
+1. Ensure that the pod identity preview is registered in your subscription by entering the command below
+
+   ```bash
+   az feature show --name EnablePodIdentityPreview \
+   --namespace Microsoft.ContainerService -o table
+   ```
+
+   ![pod identity feature verified](pictures/pod-identity-feature-verified.PNG)
+
+1. Obtain credentials to access your cluster and verify that you have access
+
+   ```bash
+   az aks get-credentials -g $RESOURCE_GROUP \
```

Comment:
Again, could be removed since they will have already done this.

Reply:
@rickrain  they need this since they just enabled pod-identity